bool isCmd
int32 direction
int32 cmdId
string graphName
float64 speed
